### ISSUE TITLE
Notifications do not always show legitimate mentions

### ISSUE THREAD

**Issue by 2000c5** [bitclout/backend#51]
Through extensive testing I am sure that the mentions are missed on notifications many times.

Have not found a rhyme or reason as to why they are not indexed.

Have confirmed with @maebeam that I am not Grey or Blacklisted.

Can see the mentions on the blockchain data but they don't show on notifications on local node or Bitclout.com.

This will surely effect user experience, so just trying to help by mentioning it.

**Comment by 2000c5** [comment]
Have confirmed other user's posts do not show in mentions as well.

First thought it was from extensive tagging, but after research I do not think is the case.

It does seem to be isolated to certain users or posts, though.

Will keep looking for the rhyme and reason...

**Comment by maebeam** [comment]
On the block explorer you can see `AffectedPublicKeys`. If someone should have received a notification for a post and their key is missing from `AffectedPublicKeys` that's a problem. Perhaps try starting there?

**Comment by 2000c5** [comment]
Thank you for replying.

I did a quick search on one of the last mentions I noticed were missing.

(This extends to multiple accounts and multiple users who mentions never show, or sometimes users normally show up but some notifications are missed)

From: ClementBourcart  BC1YLgmMPiVu3B1SkMcpHuGpZm5maWqUWdjtR8EruZyfqFdv4YYphis
Menton: RemindMe BC1YLiDHaq9FuvZYiDzXe2z4jbnphbZa7C8cGSL147HWLYHDm9Vea2Y 

PostHashHex: 70651c5cfef81f5c801ce7309d67e5698a8273a15e4e0eeae62d018c30f741f7

There are no AffectedKeys on this transaction in the block. (Height: 35772)

I could supply many more if needed.

```
{
  "BlockHashHex": "000000000000b6695eb5c5efeeca7a799b4d206829c9d0baac239431ef092dbb",
  "TxnIndexInBlock": 216,
  "TxnType": "SUBMIT_POST",
  "TransactorPublicKeyBase58Check": "BC1YLgmMPiVu3B1SkMcpHuGpZm5maWqUWdjtR8EruZyfqFdv4YYphis",
  "AffectedPublicKeys": [
    {
      "PublicKeyBase58Check": "BC1YLgmMPiVu3B1SkMcpHuGpZm5maWqUWdjtR8EruZyfqFdv4YYphis",
      "Metadata": "BasicTransferOutput"
    },
    {
      "PublicKeyBase58Check": "BC1YLhhjeJ3t4EZSkK5w257F2vxuacR4ZSbZL4Qu3axJef495PkP3gQ",
      "Metadata": "ParentPosterPublicKeyBase58Check"
    }
  ],
  "TxnOutputs": [
    {
      "PublicKey": "ApAkayhty2LpY7xaOurKoMMtT7Vnv8D1yJg63wfVecpm",
      "AmountNanos": 3747
    }
  ],
  "BasicTransferTxindexMetadata": {
    "TotalInputNanos": 4089,
    "TotalOutputNanos": 3747,
    "FeeNanos": 342,
    "UtxoOpsDump": ""
  },
  "BitcoinExchangeTxindexMetadata": null,
  "CreatorCoinTxindexMetadata": null,
  "CreatorCoinTransferTxindexMetadata": null,
  "UpdateProfileTxindexMetadata": null,
  "SubmitPostTxindexMetadata": {
    "PostHashBeingModifiedHex": "70651c5cfef81f5c801ce7309d67e5698a8273a15e4e0eeae62d018c30f741f7",
    "ParentPostHashHex": "26cc940acab9e9d1ec1621f40bd8b8cac3e433e58340297bd2c843d4b11f18b2"
  },
  "LikeTxindexMetadata": null,
  "FollowTxindexMetadata": null,
  "PrivateMessageTxindexMetadata": null,
  "SwapIdentityTxindexMetadata": null
}

```

**Comment by 2000c5** [comment]
If you want me to PM you a way to check your own mentions against your notifications - let me know. Might help for you guys to see how prevalent it is.

**Comment by 2000c5** [comment]
d33e9cbf9034e11335e7bb575de962e16204e931b8bc6992b8df21cd8f2fd3bf has AffectedKeys, but none for @RemindMe which was mentioned. (BC1YLiDHaq9FuvZYiDzXe2z4jbnphbZa7C8cGSL147HWLYHDm9Vea2Y)

```
{
  "BlockHashHex": "00000000000029eb014d005d534793fa9de370534292ef1c15d22ae0b0379fdb",
  "TxnIndexInBlock": 332,
  "TxnType": "SUBMIT_POST",
  "TransactorPublicKeyBase58Check": "BC1YLfyWj3NaC2WC3zqvz42KiVC4np9vouf6Zp5zs9HA77HTEv58YUu",
  "AffectedPublicKeys": [
    {
      "PublicKeyBase58Check": "BC1YLfyWj3NaC2WC3zqvz42KiVC4np9vouf6Zp5zs9HA77HTEv58YUu",
      "Metadata": "BasicTransferOutput"
    },
    {
      "PublicKeyBase58Check": "BC1YLfzL4FNZJ8k1xCVvVAA3UeMoQApG9V7VW1aVW9yAQdRNzHfqSWg",
      "Metadata": "MentionedPublicKeyBase58Check"
    },
    {
      "PublicKeyBase58Check": "BC1YLfzL4FNZJ8k1xCVvVAA3UeMoQApG9V7VW1aVW9yAQdRNzHfqSWg",
      "Metadata": "RecloutedPublicKeyBase58Check"
    }
  ],
  "TxnOutputs": [
    {
      "PublicKey": "AigPG5pIpXKAwu7qZigM8EWA9gGS6WXULVfZ+PcI69X+",
      "AmountNanos": 13030505
    }
  ],
  "BasicTransferTxindexMetadata": {
    "TotalInputNanos": 13031140,
    "TotalOutputNanos": 13030505,
    "FeeNanos": 635,
    "UtxoOpsDump": ""
  },
  "BitcoinExchangeTxindexMetadata": null,
  "CreatorCoinTxindexMetadata": null,
  "CreatorCoinTransferTxindexMetadata": null,
  "UpdateProfileTxindexMetadata": null,
  "SubmitPostTxindexMetadata": {
    "PostHashBeingModifiedHex": "d33e9cbf9034e11335e7bb575de962e16204e931b8bc6992b8df21cd8f2fd3bf",
    "ParentPostHashHex": ""
  },
  "LikeTxindexMetadata": null,
  "FollowTxindexMetadata": null,
  "PrivateMessageTxindexMetadata": null,
  "SwapIdentityTxindexMetadata": null
}
```

**Comment by 2000c5** [comment]
Here is one more example missing from another account's mentions, and then I'll leave this be unless asked for more info. 😁

From: MLefebvre - BC1YLh2TefSmULToFzrmXGpAcjKPJwCabU7XHTw6W63XBURSHZQZik8
Mention: CloutDash - BC1YLgD76rZNujWj5vjFhdPYzLgQWQJifh2JJbc17kYz3vgJCbT1x4i

The poster, MLefebvre is tagged as an AffectedPublicKey, as is the original poster, but not CloutDash.

```

{
  "BlockHashHex": "0000000000009ed456ddb935ff279a644c168779ed612009ed6ecdd2f19c2b77",
  "TxnIndexInBlock": 1805,
  "TxnType": "SUBMIT_POST",
  "TransactorPublicKeyBase58Check": "BC1YLh2TefSmULToFzrmXGpAcjKPJwCabU7XHTw6W63XBURSHZQZik8",
  "AffectedPublicKeys": [
    {
      "PublicKeyBase58Check": "BC1YLh2TefSmULToFzrmXGpAcjKPJwCabU7XHTw6W63XBURSHZQZik8",
      "Metadata": "BasicTransferOutput"
    },
    {
      "PublicKeyBase58Check": "BC1YLiU9Wpk8c7pVx7GCu8fae6AQhdUrGYcWc1Wz9V56HbCi7RPr887",
      "Metadata": "ParentPosterPublicKeyBase58Check"
    }
  ],
  "TxnOutputs": [
    {
      "PublicKey": "ArJyUtDvdyN3jyu9iLbzUl+56/Ue9WK78CUTrFoy054T",
      "AmountNanos": 1650
    }
  ],
  "BasicTransferTxindexMetadata": {
    "TotalInputNanos": 1997,
    "TotalOutputNanos": 1650,
    "FeeNanos": 347,
    "UtxoOpsDump": ""
  },
  "BitcoinExchangeTxindexMetadata": null,
  "CreatorCoinTxindexMetadata": null,
  "CreatorCoinTransferTxindexMetadata": null,
  "UpdateProfileTxindexMetadata": null,
  "SubmitPostTxindexMetadata": {
    "PostHashBeingModifiedHex": "17e7ab69ef66295d3cc8af3f88b6c216228c1cf81eeb6a49f9bb0c335f7209cc",
    "ParentPostHashHex": "c3067a29a4d4ddcc1a5563538ecc26becc87570bde50d42461ab66016b001b5d"
  },
  "LikeTxindexMetadata": null,
  "FollowTxindexMetadata": null,
  "PrivateMessageTxindexMetadata": null,
  "SwapIdentityTxindexMetadata": null
}
```

**Comment by stephenjones17** [comment]
I've also noticed this issue. Tested around it a little yesterday, and it seemed to be that if you have some extra characters after the @ mention — e.g. "@maebeam's" instead of "@maebeam" — then no notification will get sent. In the attached example, I didn't get a notification for the mention, but on other posts/replies where it was just the username with no text around it, it worked fine.

<img width="603" alt="Screenshot 2021-06-22 at 11 12 36 am" src="https://user-images.githubusercontent.com/86295575/122907229-d042a000-d34a-11eb-8546-c00e7bec170a.png">
